### ISSUE TITLE
[Story] Show users that reacted 

### ISSUE THREAD

**Issue by VolkerJunginger** [vector-im/element-meta#1299]
As a user that has a message w. reactions in their timeline I want to reacted so that I can see who cares.

## Scope

```[tasklist]
### Engineering
- [ ] https://github.com/vector-im/element-x-ios/issues/784
- [ ] https://github.com/vector-im/element-x-android/issues/754
- [ ] https://github.com/vector-im/element-x-ios/pull/1158
```

## Out of scope

- Nothing at the moment

### Android
- [x] Design sign-off on completion
- [x] QA sign-off on completion
- [x] Product sign-off on completion

### iOS
- [x] Design sign-off on completion
- [x] QA sign-off on completion
- [x] Product sign-off on completion

**Comment by aringenbach** [comment]
Hello @VolkerJunginger 

Couple questions around showing latest reaction in the Room list. Currently in EX, the room list just shows the latest message and not any reaction. If I understand correctly we should display these reactions when it makes sense. These are the different use cases:

1. Last message has no reaction => Show message
2. Last message is from __current user__ and has reactions => Show latest reaction (Q: does this include self-reaction with something like "You reacted with "👍" ?)
3. Last message is from __another user__ and has reactions => Not sure what we do in that case as we could do a bunch of different things listed below: 
    - Show message
    - Show __current user__ latest reaction, if they reacted, otherwise show message
    - Do the same thing as 2.
    
I haven't found all the designs for these use cases so maybe some input from design might be needed as well.

**Comment by VolkerJunginger** [comment]
Current user = Me. If I as a user...

2. Last message is from current user and has reactions => Show latest reaction 
Q: does this include self-reaction with something like "You reacted with "👍" ? -> Yes.

3. Last message is from another user and has reactions: Show message

**So we only show reactions in case:**
- My message is the last in the room and someone reacted on my message.
- I reacted on the last message in the room

We ignore reactions on other users messages (except I reacted)
